Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/BaseWindowsMixedRealityXRSDKSource.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.XRSDK.Input;
+using UnityEngine;
+using UnityEngine.XR;
+
+#if WMR_ENABLED
+using Unity.XR.WindowsMR;
+#endif // WMR_ENABLED
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.WindowsMixedReality
+{
+    /// <summary>
+    /// A Windows Mixed Reality source instance.
+    /// </summary>
+    public abstract class BaseWindowsMixedRealityXRSDKSource : GenericXRSDKController
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        protected BaseWindowsMixedRealityXRSDKSource(TrackingState trackingState, Handedness sourceHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+                : base(trackingState, sourceHandedness, inputSource, interactions) { }
+
+        private Vector3 currentPointerPosition = Vector3.zero;
+        private Quaternion currentPointerRotation = Quaternion.identity;
+        private MixedRealityPose currentPointerPose = MixedRealityPose.ZeroIdentity;
+
+        /// <summary>
+        /// Update spatial pointer and spatial grip data.
+        /// </summary>
+        protected override void UpdatePoseData(MixedRealityInteractionMapping interactionMapping, InputDevice inputDevice)
+        {
+            Debug.Assert(interactionMapping.AxisType == AxisType.SixDof);
+
+            base.UpdatePoseData(interactionMapping, inputDevice);
+
+            // Update the interaction data source
+            switch (interactionMapping.InputType)
+            {
+#if WMR_ENABLED
```

Comment:
Out of curiousity is there some case in the future where you think there's gonna be stuff outside of WMR_ENABLED? 

Or should this entire function just be wrapped in WMR_ENABLED so that we don't call an overload when not necessary.

Reply:
Hmm...I can't think of anything specific offhand for what might be outside the WMR_ENABLED in the future. I'll move it to wrap everything

Reply:
Adjusted WMR_ENABLED placement